numpy/polynomial/chebyshev.py

```diff
@@ -1663,6 +1663,130 @@ def chebfit(x, y, deg, rcond=None, full=False, w=None):
     return pu._fit(chebvander, x, y, deg, rcond, full, w)
 
 
+def chebfit2d(x, y, z, deg, rcond=None, full=False, w=None, max_degree=None):
```

Comment:
Any reason not to immediately go to nd?
```suggestion
def chebfitnd(xs, fxs, deg, *, rcond=None, full=False, w=None, max_degree=None):
```

Reply:
The function call is slightly different. The 2d version has seperate paramaters for each dimension x and y. While the nd version only has 1 parameter (x, y) with all coordinates in a tuple.
The 2d version is more consistent with the existing functions for e.g polyval2d.

Reply:
So, I'd argue that in hindsight the 2d functions were a mistake - we should have gone straight from 1d to nd, rather than blowing up our API with intermediate integers.

Since what you're adding is a new API, what I'm arguing is that we should learn from our mistake, and go straight to writing the nd function.

Reply:
If you feel strongly about the calling convention, you could use instead:
```
def chebfitnd(*args, *, deg, rcond=None, full=False, w=None, max_degree=None):
    *xs, fxs = args
```

Reply:
I have no particular attachement to the calling convention

Reply:
Why not deprecating all the XXX2d and XXX3d functions (like chebfit2d, chebfit3d, chebval2d, chebval3d, chebgrid2d,    chebgrid3d, chebvander2d,   chebvander3d, etc.) since they all are obsolete once you have the XXXnd functions (like  chebfitnd,  chebvalnd, chebgridnd, chebvandernd, ...)?